### PR TITLE
Refactor Swift package for use with an app extension

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**
I'm going to mark this as a draft b/c it's Friday at 4pm and I have more testing that I want to do *but* I wanted to give folks a warning that this is happening. :)

@dephillipsmichael
Please take a look and let me know if this mucks with anything you were intending to do for DIAN. It basically pulls out all the upload code into a different library (with far fewer dependencies).

### CODE REVIEW

**Review by Erin-Mounts — APPROVED** [review]
Looks good (including the kotlin stuff, as far as I can tell)

File: bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/DatabaseDriverFactory.kt [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
+import org.sagebionetworks.bridge.kmm.shared.IOSBridgeConfig
+import platform.Foundation.NSFileManager
 
 actual class DatabaseDriverFactory: DbDriverFactory {
     actual override fun createDriver(): SqlDriver {
-        return NativeSqliteDriver(BridgeResourceDatabase.Schema, "resource.db")
+
+        val schema = BridgeResourceDatabase.Schema
+        val name = "resource.db"
+        val path = IOSBridgeConfig.appGroupIdentifier?.let {
```
Comment (nategbrown9) [inline bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/DatabaseDriverFactory.kt:15]: Is appGroupIdentifier null for Mobile Toolbox app?

File: bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/DatabaseDriverFactory.kt [hunk 2]
```diff
@@ -1,10 +1,39 @@
...
+            NSFileManager.defaultManager.containerURLForSecurityApplicationGroupIdentifier(it)?.absoluteString
+        }
+
+        if (path == null) {
+            return NativeSqliteDriver(schema, name)
+        }
+        else {
+            val configuration = DatabaseConfiguration(
```
Comment (nategbrown9) [inline bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/DatabaseDriverFactory.kt:26]: Should we add a comment here as a reminder to check this when we update SqlDelight?

**Review by nategbrown9 — APPROVED** [review]
Looks good. Just a couple comments/questions.

File: bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/DatabaseDriverFactory.kt (hunk 1 above)
Comment (syoung-smallwisdom) [reply bridge-client/src/iosMain/kotlin/org/sagebionetworks/bridge/kmm/shared/cache/DatabaseDriverFactory.kt:15]: yes

**Comment by syoung-smallwisdom** [comment]
Closing. Will reopen with the refactoring in a separate PR

**Pull request closed without merging**